src/VirtualClient/VirtualClient.Actions/StressAppTest/StressAppTestExecutor.cs

```diff
@@ -0,0 +1,270 @@
+
+                EventContext telemetryContext = EventContext.Persisted()
+                    .AddContext("command", pathToExe)
+                    .AddContext("commandArguments", commandLineArguments);
+
+                await this.Logger.LogMessageAsync($"{nameof(StressAppTestExecutor)}.ExecuteProcess", telemetryContext, async () =>
+                {
+                    using (IProcessProxy process = this.systemManagement.ProcessManager.CreateProcess(
+                        pathToExe,
+                        commandLineArguments,
+                        workingDirectory))
+                    {
+                        this.CleanupTasks.Add(() => process.SafeKill());
+
+                        await process.StartAndWaitAsync(cancellationToken)
+                            .ConfigureAwait(false);
+
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            this.Logger.LogProcessDetails<StressAppTestExecutor>(process, telemetryContext);
+                            // process.ThrowIfErrored<WorkloadException>(ProcessProxy.DefaultSuccessCodes, errorReason: ErrorReason.WorkloadFailed);
+                        }
+                    }
+                }).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Logs the StressAppTest workload metrics.
+        /// </summary>
+        private void LogStressAppTestOutput(
+            DateTime startTime,
+            DateTime endTime,
+            string commandLineArguments,
+            string resultsFileName,
+            EventContext telemetryContext,
+            CancellationToken cancellationToken)
+        {
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                string resultsPath = this.PlatformSpecifics.Combine(this.PackageDirectory, resultsFileName);
+                if (!this.fileSystem.File.Exists(resultsPath))
+                {
+                    throw new WorkloadResultsException(
+                        $"The StressAppTest results file was not found at path '{resultsPath}'.",
+                        ErrorReason.WorkloadResultsNotFound);
+                }
+
+                string rawText = this.fileSystem.File.ReadAllText(resultsPath);
+
+                if (string.IsNullOrWhiteSpace(rawText))
+                {
+                    throw new WorkloadResultsException(
+                        "The StressAppTest workload did not produce valid results. The results file is blank",
+                        ErrorReason.WorkloadResultsNotFound);
+                }
+
+                StressAppTestMetricsParser parser = new StressAppTestMetricsParser(rawText);
+                IList<Metric> workloadMetrics = parser.Parse();
+
+                foreach (Metric metric in workloadMetrics)
+                {
+                    telemetryContext
+                        .AddContext("testRunResult", metric.Tags[0] ?? string.Empty);
```

Comment:
Just curious, why do these test RunResults need to be in tags, are these just boolean of true and false?

Reply:
The test RunResults have additional data like in which DIMM slot was the error detected, etc. It is being added to help the user to debug the issues when used on a system